### DISCUSSION TITLE
Tokens + Two factor authentication

### DISCUSSION THREAD

**Discussion by KevinMusgrave** [community/community#29220] [Other Feature Feedback, Questions, & Ideas] [answered]
When pulling / pushing code via the command line, I use a token. If a malicious actor obtained my token, they would be able to push and pull code using my account. Two-factor authentication would prevent this.

Can we have the option to require two-factor authentication when using tokens?

**Comment by airtower-luna** [comment, marked as answer]
> Can we have the option to require two-factor authentication when using tokens?

Seems unlikely, it'd be hard to do automation with Git over HTTPS otherwise (at least without major changes to all clients). And the token requirement was [introduced somewhat recently](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/), it's mandatory since about a year ago. :sweat_smile: 

For stronger authentication, look at [using Git over SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh). That way you do public key authentication (your private key never needs to leave your machine) instead of a token that's basically a shared secret. The key file can be stored encrypted, or you can even use a key stored in a hardware token (the `sk-*` key types).
